--- a/gdb/darwin-nat.c	2019-02-26 14:43:44.000000000 -0500
+++ b/gdb/darwin-nat.c	2019-02-26 14:46:57.000000000 -0500
@@ -1151,7 +1151,7 @@
 			      res, wstatus);
 
 	      /* Looks necessary on Leopard and harmless...  */
-	      wait4 (inf->pid, &wstatus, 0, NULL);
+	      wait4 (inf->pid, &wstatus, WNOHANG, NULL);
 
 	      inferior_ptid = ptid_t (inf->pid, 0, 0);
 	      return inferior_ptid;
